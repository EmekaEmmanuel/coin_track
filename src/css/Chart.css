* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  letter-spacing: 0.2px;
}

.app_container {
  margin: auto;
  background-color: #111315;
}

.chart_main_tag {
  text-align: center;
}

.chart_section1 {
  padding: 50px 0;
  width: 50%;
  margin: auto;
}

.chart_font1 {
  font-size: 11px;
  color: #efefef9a;
}

.chart_font2 {
  font-size: 50px;
  color: #efefef;
  padding: 20px 0;
}

.chart_font3 {
  font-size: 11px;
  color: #efefef9a;
}

.chart_font4 {
  color: #efefef9a;
}

.button_box {
  padding: 30px 0;
  display: flex;
  justify-content: center;
  gap: 20px;
}

.chart_section2 {
  width: 90%;
  margin: auto;
  padding: 30px 30px;
  background-color: #020a11;
  border-radius: 15px;
}

.chart_section3_btn {
  padding: 8px 15px;
  background-color: #0077fe;
  border: none;
  border-radius: 10px;
}

@media only screen and (max-width: 1440px) {
  .chart_section1 {
    padding: 50px 0;
    width: 45%;
    margin: auto;
  }
}

@media only screen and (max-width: 1024px) {
  .chart_section1 {
    padding: 50px 0;
    width: 60%;
    margin: auto;
  }
}

@media only screen and (max-width: 820px) {
  .chart_section1 {
    padding: 50px 0;
    width: 70%;
    margin: auto;
  }

  .chart_font1 {
    font-size: 9px;
    color: #efefef9a;
  }

  .chart_font2 {
    font-size: 40px;
    color: #efefef;
    padding: 20px 0;
  }

  .chart_font3 {
    font-size: 10px;
    color: #efefef9a;
  }

  .chart_section2 {
    width: 95%;
    margin: auto;
    padding: 30px 30px;
    background-color: #020a11;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 425px) {
  .chart_section1 {
    padding: 30px 0 10px 0;
    width: 90%;
    margin: auto;
  }

  .chart_font2 {
    font-size: 30px;
    color: #efefef;
    padding: 20px 0;
  }

  .chart_font3 {
    display: none;
  }

  .chart_section2 {
    width: 100%;
    margin: auto;
    padding: 15px 20px;
    background-color: #020a11;
    border-radius: 15px;
  }
}

@media only screen and (max-width: 375px) {
  .chart_font2 {
    font-size: 25px;
    color: #efefef;
    padding: 20px 0;
  }
}
