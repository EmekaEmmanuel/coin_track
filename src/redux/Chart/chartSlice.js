import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
  loading: false,
  charts: [],
  error: '',
};

export const testFetchCharts = createAsyncThunk('market/fetchMarkets', async () => {
  const { data } = await (axios.get('https://api.coinstats.app/public/v1/charts?period=1m&coinId=bitcoin'));
  return data;
});

export const fetchCharts = createAsyncThunk('market/fetchMarkets', async (id) => {
  const { data } = await (axios.get(`https://api.coinstats.app/public/v1/charts?period=1m&coinId=${id}`));
  return data;
});

/* eslint-disable no-param-reassign */

const chartSlice = createSlice({
  name: 'chart',
  initialState,
  extraReducers: (builder) => {
    builder.addCase(fetchCharts.pending, (state) => {
      state.loading = true;
    })
      .addCase(fetchCharts.fulfilled, (state, action) => {
        state.loading = false;
        state.error = '';
        state.charts = action.payload.chart;
      })
      .addCase(fetchCharts.rejected, (state) => {
        state.loading = false;
        state.error = '';
      });
  },
});

/* eslint-disable no-param-reassign */

export default chartSlice.reducer;
