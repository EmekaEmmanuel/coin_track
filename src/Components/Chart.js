import React from 'react';
import { PropTypes } from 'prop-types';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js';

import { Line } from 'react-chartjs-2';
import '../css/Chart.css';

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
);

function Chart(props) {
  const { myCharts } = props;

  const newMeFunc = () => {
    const arrayPrice = [];
    myCharts?.forEach((element) => {
      arrayPrice.push(element[1]);
    });
    return { arrayPrice };
  };
  const newMe = newMeFunc();

  const options = {
    maintainAspectRatio: false,
    scales: {
    },
    title: {
      display: true,
      text: 'Users Gained between 2016-2020',
    },
    legend: {
      labels: {
        fontSize: 25,
      },
    },
  };

  const data = {
    // eslint-disable-next-line max-len
    labels: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31],
    datasets: [
      {
        label: 'Base Point changes',
        data: newMe.arrayPrice,
        backgroundColor: [
          'rgba(255, 99, 132, 0.2)',
          'rgba(54, 162, 235, 0.2)',
          'rgba(255, 206, 86, 0.2)',
          'rgba(75, 192, 192, 0.2)',
          'rgba(153, 102, 255, 0.2)',
          'rgba(255, 159, 64, 0.2)',
        ],
        borderColor: [
          'rgba(255, 99, 132, 1)',
          'rgba(54, 162, 235, 1)',
          'rgba(255, 206, 86, 1)',
          'rgba(75, 192, 192, 1)',
          'rgba(153, 102, 255, 1)',
          'rgba(255, 159, 64, 1)',
        ],
        borderWidth: 1,
      },
    ],
  };

  return (
    <div className="app_container">
      <main className="chart_main_tag">

        <section className="chart_section1">
          <p className="chart_font1">LIVE MARKET CHANGES</p>
          <h1 className="chart_font2">Get an Overview on the Live chart</h1>
          <p className="chart_font3">Your favorite place to see moment changes.</p>
        </section>

        <section className="chart_section2">
          <article className="chart_article">

            <h3 className="chart_font4">Live chart</h3>

            <div>
              <Line
                data={data}
                height={400}
                options={options}
              />
            </div>

            <div className="button_box">
              <button className="chart_section3_btn" type="button">1 month</button>
            </div>

          </article>
        </section>
      </main>
    </div>
  );
}

Chart.propTypes = {
  myCharts: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.object)).isRequired,
};

export default Chart;
