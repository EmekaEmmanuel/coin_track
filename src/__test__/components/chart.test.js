import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Chart from '../../Components/Chart';

it('Chart component renders correctly', () => {
  const myCharts = store.getState().chart.charts;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Chart myCharts={myCharts} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
