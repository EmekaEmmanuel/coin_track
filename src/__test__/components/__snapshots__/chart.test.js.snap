// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component renders correctly 1`] = `
<div
  className="app_container"
>
  <main
    className="chart_main_tag"
  >
    <section
      className="chart_section1"
    >
      <p
        className="chart_font1"
      >
        LIVE MARKET CHANGES
      </p>
      <h1
        className="chart_font2"
      >
        Get an Overview on the Live chart
      </h1>
      <p
        className="chart_font3"
      >
        Your favorite place to see moment changes.
      </p>
    </section>
    <section
      className="chart_section2"
    >
      <article
        className="chart_article"
      >
        <h3
          className="chart_font4"
        >
          Live chart
        </h3>
        <div>
          <canvas
            height={400}
            role="img"
            width={300}
          />
        </div>
        <div
          className="button_box"
        >
          <button
            className="chart_section3_btn"
            type="button"
          >
            1 month
          </button>
        </div>
      </article>
    </section>
  </main>
</div>
`;
